### PR TITLE
sdl: add version 2.0.18

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **sdl/2.0.18**

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by madebr — APPROVED** [review]

**Review by edoren — APPROVED** [review]

File: recipes/sdl/all/conanfile.py [hunk 1]
```diff
@@ -150,6 +150,10 @@ def validate(self):
         if self.settings.os == "Macos" and not self.options.iconv:
             raise ConanInvalidConfiguration("On macOS iconv can't be disabled")
 
+        # SDL>=2.0.18 requires xcode 12 or higher because it uses CoreHaptics.
+        if tools.Version(self.version) >= "2.0.18" and tools.is_apple_os(self.settings.os) and tools.Version(self.settings.compiler.version) < "12":
```
Comment (SSE4) [inline recipes/sdl/all/conanfile.py:154]: that's really weird, CoreGraphics was available very long before Xcode 12...

**Review by SSE4 — APPROVED** [review]

File: recipes/sdl/all/conanfile.py (hunk 1 above)
Comment (toge) [reply recipes/sdl/all/conanfile.py:154]: It's not CoreGraphis. 
CoreHaptics is introduced to support haptic feedback of game controller.
Please refer official document. https://developer.apple.com/documentation/corehaptics/playing_haptics_on_game_controllers

Comment (uilianries) [reply recipes/sdl/all/conanfile.py:154]: It's [CoreHaptics](https://developer.apple.com/documentation/corehaptics). According the link it says +13, but I have no idea.

**Review by uilianries — APPROVED** [review]

File: recipes/sdl/all/conanfile.py (hunk 1 above)
Comment (SSE4) [reply recipes/sdl/all/conanfile.py:154]: oh, yes, I was reading too fast xD

**Pull request merged**